src/libraries/Directory.Build.props

```diff
@@ -84,7 +84,10 @@
     <StrongNameKeyId Condition="$(MSBuildProjectName.StartsWith('Microsoft.Extensions.'))">MicrosoftAspNetCore</StrongNameKeyId>
     <!-- We can't generate an apphost without restoring the targeting pack. -->
     <UseAppHost>false</UseAppHost>
-    <EnableDefaultItems>false</EnableDefaultItems>
+    <!-- TODO: Enable default items in test projects. -->
+    <EnableDefaultItems Condition="'$(IsTestProject)' == 'true' or
```

Comment:
hmm, I thought we were pretty selective about source inclusions dependent on platforms (say with *.Unix.cs files and all), say with System.Private.Corelib. Are we sure this currently works? Or is there also some massaging in this PR to do that.

Reply:
The third commit handles updating all projects to work with that setting. Less than 50% of our projects multi-target based on the OS, the majority doesn't. For the projects that don't multi-target based on the OS, this settings work nicely.

For the ones that multi-target based on the OS, this setting still works fine but a bit more work was required to make sure that nothing unexpected is included which previously wasn't. I recommend to look into the third commit yourself as the diff mostly speaks for itself.

Reply:
FWIW, I really, really, really (insert a string of "really"s until your ears bleed) do not want us to use EnableDefaultItems.

I don't even want the SDK to support it.

It's bad engineering practice that's only good for bootstrapping CS101 students.

Reply:
I agree with Jeremy.  If the overwhelming majority of others disagree, I'll acquiesce, but understanding exactly what's being included and what's not being included is really helpful.  And with the variations in our builds and our need to include some files in some configurations and not in others, we're never going to get to a place where everything is just always included, which from my perspective also means the consistent thing is always specifying exactly what should be in every build.

Reply:
(Also, for fundamental overhauls like this, that impact everyone working on the project, that impact every project, etc., I'd prefer if we have a discussion about the direction rather than an ~500 file diff just appearing.  If it was necessary for you to do the work in order to come up with the proposal, then ok, and it being potentially throw-away is just part of the process; but if that could have been achieved just by experimenting with a couple of projects, presumably that would have resulted in less work for everyone?)

Reply:
My thinking is that the lower-level your code is, the more you are writing platform-specific code. And the more platform-specific code you write, the more you need to be explicit about what is and what isn't included in your project. I don't think it has anything to do with size of your project. dotnet/aspnetcore is by no means "small" or CS101, but they use EnableDefaultItems in all their projects.

The same can be said for using `#if`s in your code. If you only need them in one or two spots, it is fine. But at the level we need to specialize code in the shared fx, it would be a nightmare.

That's why I think the current situation we have is acceptable. We have code that is both low-level (shared fx) and high-level (Microsoft.Extensions). Using the approach that best fits the project makes sense to me.